doc/designs/fido2.md

```diff
@@ -0,0 +1,239 @@
+# FIDO2 authentication
+
+## Overview
+
+Traditional authentication with a password is not considered secure enough
+by many companies or government agencies. Alternate and more secure
+solutions exist, among which the use of FIDO2 devices, where the private
+key is stored on the device and the server only needs to know the public
+key.
+
+The goal of this feature is to use a FIDO2 device to authenticate a user
+against IPA.
+
+The project will be jointly developed by SSSD and IPA:
+
+- IPA provides the interface to store the user's public credentials
+- IPA provides the interface to configure FIDO2 settings
+- SSSD performs the actual authentication
+
+## Use Cases
+
+- The administrator or the user registers a FIDO2 device into IPA, associated
+to a user account. The registration requires the insertion of the key, or
+the string output provided by SSSD registration mechanism.
+- The user is then able to authenticate to any IPA enrolled host using the
+FIDO2 device. Authentication methods include console login, desktop login,
+or su access.
+
+## How to Use
+
+### Configuration of the FIDO2 settings by the administrator
+
+The administrator is able to specify common settings that will apply:
+
+- request user verification during authentication (On/Off/Default):
+  - On: request user verification during authentication (PIN for instance).
+  - Off: do not request user verification during authentication.
+  - Default: fallback to the FIDO2 device’s default behavior.
```

Comment:
Is it request or require?

Reply:
I tried to update all places with "require", hope I didn't miss any.